M2/Macaulay2/packages/IncidenceCorrespondenceCohomology.m2

```diff
@@ -0,0 +1,2105 @@
+		if (a%p^k)-a0==(p^k -p)/2 then( 
+		    -- similarly  b=bk2*p^k2+.... b1*p+a0, we check that bj=(p-1)/2 for j=1,...,k-1
+		    if (b%p^k)-b0== (p^k -p)/2 then(
+			ak:= (a%p^(k+1))//p^k;
+			bk:= (b%p^(k+1))//p^k;
+			if ak+bk<p then (
+			    if k1==k2 then true 
+			    else  if( k1<k2 and ak<= bk) or (k2<k1 and bk<= ak) then true
+			    else false)---if any of the condition above fails return false
+		      	else false)                   
+                    else false)               
+                else false)             
+            else false)         
+        else false)
+
+
+--User function: This function checks if the monomial complete intersection (ZZ/p)[x_1,...,x_n]/(x_1^{a_1},...,x_n^{a_n})  has the Strong Lefschetz Property (SLP) in characteristic p 
+--This function uses the classification presented in Theorem 3.8 [Lundqvist, Nicklasson'18], or, when the option UseConjecture is set to be true, the Han-Monsky multiplication adopting Conjecture 4.1 [KMRR,25]
+--input: (p, L) where p is prime and L={a_1,a_2 ... a_n} is a list of integers 
+--output: true if  k[x_1,...,x_n]/(x_1^{a_1}, …, x_n^{a_n}) has the SLP in characteristic p, false otherwise
+
+
+hasSLP = method(Options => {UseConjecture => false})
+
+hasSLP(ZZ, List) := opts -> (p, L) -> (
+    n:=#L;
+    if n <= 1 or p == 0 then true else (
+    	if sum L - n < p then true 
+	else (
+    	    if not(opts.UseConjecture) then ( 
+		if p==2 then ( 
+		    if n==2 and ((L#0==2 and  L#1%2==1)  or (L#1==2 and  L#0%2==1) or 
+			(L#0==3 and  L#1%4==2)  or (L#0==3 and  L#1%4==2))   then true	 -- by Corollary 6.3. [Cook'11]  
+		    else false)-- by Corollary 4.8  [Cook'11] if n>2 and p=2 we don't have the SLP
+		else( L1:=select(L, x -> x>=p);-- list the a_j bigger then p
+		    if #L1!=1 then  ( 
+			if #L==2 and #L1==2 then (return auxSLP2(p, L#0, L#1);)--if this is the case when n=2, p>2 and both a_1,a_2>=p, so we use auxSLP
+			else  false)-- if n>2 it is necessary that exactly one a_j is bigger then p
+		    else (
+			d:=L1#0;
+			d0:=d%p;
+			if sum L -n-(d-1)<=min{p-d0,d0} then true 
+			else  false
+			    )
+			)
+		    )
+	     else ( -- This case uses Conjecture 4.1 [KMRR,25] for the Han-Monsky multiplication
+		H := recSLP(p,L);
+	        not(H === false)
+            	)
+             )
+    	 )
+     )
+
+
+
+
+
+-----------------------------------------------------------
+-----------------------------------------------------------
+--Computations of sheaf cohomology for divided powers of the cotangent sheaf on projective space
+-----------------------------------------------------------
+-----------------------------------------------------------
+
+--Input: prime integer p, and list of non-negative integers less than p list1
+--list1 corresponds to a positive integer written in base p
+--Output: the corresponding integer now in base 10
+basep10 = (p, list1) ->(
+    if list1 == {}
+    then return 0
```

Comment:
I don't think you need the special case for the empty list here, since the `for` loop on the next line will return an empty list in this case, and `sum {}` will evaluate to 0 already.